commit 15bf0047eb84575ef792e4391fc060be9344b210
Author: Mike Smith <grimbough@gmail.com>
Date:   Wed Mar 8 22:26:01 2023 +0100

    lz4 printing

diff --git a/src/compression_tools/blosc/lib/lz4-1.9.2/lz4.c b/src/compression_tools/blosc/lib/lz4-1.9.2/lz4.c
index 9808d70..c83c848 100644
--- a/src/compression_tools/blosc/lib/lz4-1.9.2/lz4.c
+++ b/src/compression_tools/blosc/lib/lz4-1.9.2/lz4.c
@@ -228,9 +228,9 @@ static const int LZ4_minLength = (MFLIMIT+1);
 static int g_debuglog_enable = 1;
 #  define DEBUGLOG(l, ...) {                                  \
                 if ((g_debuglog_enable) && (l<=LZ4_DEBUG)) {  \
-                    fprintf(stderr, __FILE__ ": ");           \
-                    fprintf(stderr, __VA_ARGS__);             \
-                    fprintf(stderr, " \n");                   \
+                      REprintf( __FILE__ ": ");           \
+                      REprintf( __VA_ARGS__);             \
+                      REprintf(" \n");                   \
             }   }
 #else
 #  define DEBUGLOG(l, ...)      {}    /* disabled */
